### ISSUE TITLE
Dependencies

### ISSUE THREAD

**Issue by srudkin12** [asheshrambachan/HonestDiD#28]
This is an excellent package and really helpful readme. Easy to follow and the results replicated quickly on my system.

When following the readme I was asked to install tidyverse and Rglpk  but these are not mentioned in the readme as being needed.

**Comment by jonathandroth** [comment]
@mcaceresb can you look into this dependency issue? Thanks!

**Comment by mcaceresb** [comment]
@jonathandroth I added  `Rglpk` as an explicit dependency in #23. A few packages have `lifecycle` as a dependency, which in turn has `tidyverse` as a dependency.

@srudkin12 `tidyverse` is only needed to the extent other dependencies need it. `Rglpk` is used directly and I added it as a dependency, but you shouldn't need to load it as a library separately after installing and loading HonestDiD.

**Comment by mcaceresb** [comment]
@jonathandroth @srudkin12 Actually I found another issue: I removed `tidyverse` from the `library` statements in the code in #23. I don't think any of the library statements are needed in general, but I hadn't noticed it tried to still load tidyverse.

**Comment by grantmcdermott** [comment]
Hey folks, I just had a look at this package and jumped on here to mention this same issue.

More generally, it's a big no-no to include _any_ library statements at the top of internal R package scripts. These should all be removed. I can see that Mauricio started in #23, but the rest of them will have to go too. ([Example](https://github.com/asheshrambachan/HonestDiD/blob/master/R/sensitivityresults.R#L11-L19).) 

A lazy option would be to move these packages to Depends in the DESCRIPTION file, so that they auto-load along with this package. However, that too is not considered good practice and should only be done as holdover / last resort IMO.

Better would be to use explicit namespace assignment when you call a function, e.g. `foreach::foreach`... which I think you are already doing and so may not require any further work? I don't think you have a unit testing framework set up, but you should be able to run `devtools::check()` / `R CMD check` pretty quickly to at least test the documentation examples.

(You can also use [`importFrom`](https://r-pkgs.org/dependencies-in-practice.html) statements as part of your docs, but I think the `package::function` approach is the easiest here.)

**Comment by mcaceresb** [comment]
@grantmcdermott Just flagging that I've done some work addressing these (and I have a local branch passing devtools::check() / R CMD check too). I'm waiting on my other PRs to be merged but all these things you mentioned are in the pipeline for being merged! (:

**Comment by grantmcdermott** [comment]
That's great, thanks @mcaceresb. I need to play around with the package a bit more, but might have some other ideas for how to speed up / simplify some internals.

**Comment by jonathandroth** [comment]
Grant, thanks for the feedback! And Mauricio, thanks for implementing!

Are you waiting on something from me / @Ashesh Rambachan
***@***.***> re the PRs you mentioned?

**Comment by mcaceresb** [comment]
@jonathandroth No, they're ready to be merged.

**Comment by jonathandroth** [comment]
You have permissions to merge when ready, right? Or no?

**Comment by mcaceresb** [comment]
@jonathandroth No, I don't have write access.

**Comment by asheshrambachan** [comment]
@mcaceresb Just added you as a collaborator (let me know if you got it). That you should give you write access to merge in pulls.
